cmds/which/which.go

```diff
@@ -21,38 +22,33 @@ import (
 )
 
 var (
-	flags struct {
-		allPaths bool
-	}
+	allPaths = flag.Bool("a", false, "print all matching pathnames of each argument")
+	verbose  = flag.Bool("v", false, "verbose output")
 )
 
-func init() {
-	flag.BoolVar(&flags.allPaths, "a", false, "print all matching pathnames of each argument")
-}
-
-func which(p string, writer io.Writer, cmds []string) {
-	pathArray := strings.Split(p, ":")
-
-	// If no matches are found will exit 1, else 0
-	exitValue := 1
+func which(writer io.Writer, paths []string, cmds []string, allPaths bool) error {
+	var foundOne bool
 	for _, name := range cmds {
```

Comment:
and we're not using LookPath because we pass the paths in?

Reply:
I didn't know LookPath was a thing! that's pretty neat. Yea, we also have a different exec approximation.